src/rest_vol_dataset.c

```diff
@@ -4137,4 +4165,201 @@ rv_dataset_write_cb(hid_t mem_type_id, hid_t mem_space_id, void *buf, struct res
+
+            if (H5Sget_simple_extent_dims(space_id, dims, NULL) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't get dataspace dimension size");
+
+            if (NULL == (start = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*start))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'start' values");
+            if (NULL == (stride = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*stride))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'stride' values");
+            if (NULL == (count = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*count))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'count' values");
+            if (NULL == (block = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*block))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'block' values");
+
+            if (nblocks = H5Sget_select_hyper_nblocks(space_id) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't get number of hyperslab blocks");
+
+            if (H5Sget_regular_hyperslab(space_id, start, stride, count, block) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't get regular hyperslab selection");
+
+            /* For contiguous, the stride should be 1. */
+            for (i = 0; i < ndims; i++) {
+                if (stride[i] > 1)
+                    FUNC_GOTO_DONE(FALSE);
+            }
+
+            if (nblocks > 1) {
+                /* Multiple blocks: count should be 1 except for the last dimension (fastest) */
+                for (i = 0; i < ndims - 1; i++) {
+                    if (count[i] > 1)
+                        FUNC_GOTO_DONE(FALSE);
+                }
+            }
+
+            /* For contiguous, all faster running dimensions than the current dimension should be selected
+             * completely */
+            whole = (start[ndims - 1] == 0) && (count[ndims - 1] * block[ndims - 1] == dims[ndims - 1]);
+            for (i = ndims - 2; i >= 0; i--) {
+                if ((dims[i] > 1) && (count[i] * block[i] > 1) && !whole)
+                    FUNC_GOTO_DONE(FALSE);
+
+                whole = whole && (start[i] == 0) && (count[i] * block[i] == dims[i]);
+            }
+            break;
+        } /* H5S_SEL_HYPERSLABS */
+
+        case H5S_SEL_POINTS:
+            /* Assumption: any point selection is non-contiguous in memory */
+            FUNC_GOTO_DONE(FALSE);
+            break;
+
+        case H5S_SEL_ALL:
+        case H5S_SEL_ERROR:
+        case H5S_SEL_N:
+        case H5S_SEL_NONE:
```

Comment:
I'd prefer to get this PR merged, but if you could make the following changes in a future PR I'd appreciate it:

- H5S_SEL_ALL should return true for is_contiguous
- H5S_SEL_NONE should return false for is_contiguous
- H5S_SEL_ERROR and H5S_SEL_N should make this function return FAIL since it shouldn't get called with those